watertap/unit_models/tests/test_cstr_injection.py

```diff
@@ -82,237 +69,249 @@
-                        * sapon.fs.properties.cp_mol
-                        * (
-                            sapon.fs.unit.inlet.temperature[0]
-                            - sapon.fs.properties.temperature_ref
-                        )
-                    )
-                    - (
-                        sapon.fs.unit.outlet.flow_vol[0]
-                        * sapon.fs.properties.dens_mol
-                        * sapon.fs.properties.cp_mol
-                        * (
-                            sapon.fs.unit.outlet.temperature[0]
-                            - sapon.fs.properties.temperature_ref
-                        )
+        self.unit_solutions[
+            (
+                (
+                    m.fs.unit.outlet.flow_vol[0]
+                    * m.fs.properties.dens_mol
+                    * m.fs.properties.cp_mol
+                    * (
+                        m.fs.unit.outlet.temperature[0]
+                        - m.fs.properties.temperature_ref
                     )
-                    + sapon.fs.unit.control_volume.heat_of_reaction[0]
                 )
+                - m.fs.unit.control_volume.heat_of_reaction[0]
             )
-            <= 1e-3
+        ] = value(
+            m.fs.unit.inlet.flow_vol[0]
+            * m.fs.properties.dens_mol
+            * m.fs.properties.cp_mol
+            * (m.fs.unit.inlet.temperature[0] - m.fs.properties.temperature_ref)
         )
 
-    @pytest.mark.solver
-    @pytest.mark.skipif(solver is None, reason="Solver not available")
-    @pytest.mark.component
-    def test_costing(self, sapon):
-        m = sapon
+        return m
+
 
+class TestCosting_Saponification(UnitTestHarness):
+    def configure(self):
+        m = build()
+
+        # Add unit model costing
         m.fs.costing = WaterTAPCosting()
 
         m.fs.unit.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
         m.fs.costing.cost_process()
         m.fs.costing.add_LCOW(m.fs.unit.control_volume.properties_out[0].flow_vol)
-        solver = get_solver()
-        results = solver.solve(m)
+        m.objective = Objective(expr=m.fs.costing.LCOW)
 
-        assert_optimal_termination(results)
+        iscale.set_scaling_factor(m.fs.unit.costing.capital_cost, 1e-2)
+
+        iscale.calculate_scaling_factors(m.fs.unit)
+
+        self.unit_solutions[m.fs.unit.costing.capital_cost] = 7.75429 * 2
+
+        m.fs.unit.initialize()
+
+        results = solver.solve(m)
 
-        # Check solutions
-        assert pytest.approx(7.75429 * 2, rel=1e-5) == value(
-            m.fs.unit.costing.capital_cost
-        )
         assert pytest.approx(0.00082698, rel=1e-5) == value(m.fs.costing.LCOW)
 
+        return m
+
+
+class TestSaponification(object):
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_conservation(self):
+
+        sapon = build()
+
```

Comment:
Is this actually testing conservation? It seems unfinished

Reply:
deleted